internal/compiler-bridge/src/main/scala/xsbt/CompilerInterface.scala

```diff
@@ -19,16 +19,16 @@ import scala.reflect.io.AbstractFile
 import Log.debug
 import java.io.File
 
-final class CompilerInterface {
-  def newCompiler(
+final class CompilerInterface extends CompilerInterface2 {
```

Comment:
Compared to the other parts of the diff, this seems backwards.  Why does CompilerInterface(1) extends CompilerInterface2?

Reply:
`xsbt.CompilerInterface` is a hard-coded class name all bridges use as entry point.
`xsbti.CompilerInterface2` is the new Java interface.

Reply:
> xsbt.CompilerInterface is a hard-coded class name all bridges use as entry point

Now that we're coding against interfaces, we could replace the use of hardcoded class names by a file in META-INF listing the entry points of the bridge (cf https://docs.oracle.com/javase/8/docs/api/java/util/ServiceLoader.html), maybe something worth considering for a follow-up PR?

Reply:
> Now that we're coding against interfaces, we could replace the use of hardcoded class names by a file in META-INF listing the entry points of the bridge

Thanks for the pointer. That sounds like a good idea indeed.

Reply:
Ah, I see.  `CompilerInterface` was/is the source-based interface to the compiler, but it's actually implemented as a class (not an `interface`).

The new interface being defined are java `interface`s, with `CompilerInterface1` and `CompilerInterface2`, and the implementation here now implements `CompilerInterface2`.

Seeing as we're not dropping support for the older interface (the older methods, that were defined directly in this class), now captured by `CompilerInterface1`, I think it would be clearer for this to extend `CompilerInterface1` too.

Reply:
> Seeing as we're not dropping support for the older interface (the older methods, that were defined directly in this class), now captured by `CompilerInterface1`, I think it would be clearer for this to extend `CompilerInterface1` too.

Scratch that.  We don't drop support for _consuming_ the old interface (CompilerInterface1) which will continue to be supplied by current sbt-dotty (until they switch).  But the Scala 2 bridges by Zinc 1.4+ will be implementing the new, CompilerInterface2, methods.